src/into_dart.rs

```diff
@@ -92,43 +109,89 @@ impl IntoDart for CString {
     }
 }
 
-trait Num8Bit {}
+impl IntoDartExceptPrimitive for CString {}
 
-impl Num8Bit for u8 {}
+pub(crate) trait DartTypedDataTypeVisitor {
+    fn visit<T: DartTypedDataTypeTrait>(&self);
+}
 
-impl Num8Bit for i8 {}
+/// The Rust type for corresponding [DartTypedDataType]
+pub trait DartTypedDataTypeTrait {
+    fn dart_typed_data_type() -> DartTypedDataType;
 
-fn vec_8bit_to_dart<T: Num8Bit>(
-    vec: Vec<T>,
-    ty: DartTypedDataType,
-) -> DartCObject {
-    let mut vec = ManuallyDrop::new(vec);
-    let data = DartNativeTypedData {
-        ty,
-        length: vec.len() as isize,
-        values: vec.as_mut_ptr() as *mut _,
-    };
-    DartCObject {
-        ty: DartCObjectType::DartTypedData,
-        value: DartCObjectValue {
-            as_typed_data: data,
-        },
-    }
+    fn function_pointer_of_free_zero_copy_buffer() -> DartHandleFinalizer;
```

Comment:
should it be just `finalizer`? easier to pronounce, I guess 😅

Reply:
Maybe no, since it is finalizer **of ZeroCopyBuffer**, not finalizer of the type T.